addons/hr_holidays/models/hr_leave.py

```diff
@@ -788,40 +789,48 @@ def _validate_leave_request(self):
         by creating a calendar event and a resource time off. """
         holidays = self.filtered(lambda request: request.holiday_type == 'employee')
         holidays._create_resource_leave()
-        for holiday in holidays.filtered(lambda l: l.holiday_status_id.create_calendar_meeting):
-            meeting_values = holiday._prepare_holidays_meeting_values()
-            meeting = self.env['calendar.event'].with_context(no_mail_to_attendees=True, active_model=self._name).create(meeting_values)
-            holiday.write({'meeting_id': meeting.id})
+        meeting_holidays = holidays.filtered(lambda l: l.holiday_status_id.create_calendar_meeting)
+        if meeting_holidays:
+            meeting_values = meeting_holidays._prepare_holidays_meeting_values()
+            meetings = self.env['calendar.event'].with_context(
+                no_mail_to_attendees=True,
+                active_model=self._name).create(meeting_values)
+            for i in range(len(meetings)):
+                meeting_holidays[i].meeting_id = meetings[i]
```

Comment:
~~bite~~ byte shading
```suggestion
            meetings = self.env['calendar.event'].with_context(
                no_mail_to_attendees=True,
                active_model=self._name
            ).create(meeting_values)
            for i, meeting in enumerate(meetings):
                meeting_holidays[i].meeting_id = meeting
```